lib/vector-core/src/config/mod.rs

```diff
@@ -118,21 +118,21 @@ impl Output {
         }
     }
 
-    /// Set the schema definition for this output.
+    /// Set the schema definition for this `Output`.
     #[must_use]
     pub fn with_schema_definition(mut self, schema_definition: schema::Definition) -> Self {
         self.log_schema_definition = Some(schema_definition);
         self
     }
-}
 
-impl<T: Into<String>> From<(T, DataType)> for Output {
-    fn from((name, ty): (T, DataType)) -> Self {
-        Self {
-            port: Some(name.into()),
-            ty,
-            log_schema_definition: None,
-        }
+    /// Set the port name for this `Output`.
+    #[must_use]
+    pub fn with_port<S>(mut self, name: S) -> Self
+    where
+        S: Into<String>,
+    {
+        self.port = Some(name.into());
+        self
```

Comment:
nit:

```suggestion
    /// Set the port name for this `Output`.
    #[must_use]
    pub fn with_port(mut self, name: impl Into<String>) -> Self {
        self.port = Some(name.into());
        self
```

Reply:
Out of curiosity, why is this better?
And why not this?
```rust
pub fn with_post<S: Into<String>>(mut self, name: S) -> Self {
```

@001wwang might be interested by this as well

Reply:
It's not "better" (it results in the same code). I tagged it as a "nit", since I believe it to be easier to read, since it involves fewer sigils, and it makes it easier to read the code from left-to-right:

```rust
pub fn with_port(mut self, name: impl Into<String>) -> Self {
```

```rust
pub fn with_post<S: Into<String>>(mut self, name: S) -> Self {
```

```rust
pub fn with_port<S>(mut self, name: S) -> Self
where
    S: Into<String>,
{
```

But, as I said, I don't mind either way, we don't have a style guide for this, and if people prefer one over the other, that's fine by me.

Reply:
Heh, chalk this up to me having done Rust for a while before this syntax was added. It's a good nit, I'll make the switch. 

@jdrouet for a little historical background, one of the motivations for this syntax was to tidy up code that passes closures around. So, previously, you'd need something like:

```rust
fn hof(func: F) where F: Box<dyn Fn(usize) -> usize> {
...
}
```

but now you can write:

```rust
fn hof(func: impl Fn(usize) -> usize) {
...
}
```

This is _also_ quite nice for code that passes iterators around.

Reply:
Incorporated in 4e76350bc